### PR TITLE
feat(ux-tabs): create ux-tabs and ux-tab elements

### PR DESCRIPTION
**Pull request by manuel-guilbault**
linked to #83

### CODE REVIEW

**Comment by manuel-guilbault** [comment]
I also added some examples in the ux showcase app: https://github.com/manuel-guilbault/aurelia-app-ux-showcase

File: src/tabs/ux-tab-theme.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+styles.tab:hover:not(.disabled) {
+  opacity: 1;
+}
+
+styles.tab.selected:not(.disabled) {
+  opacity: 1;
+  padding-bottom: 18px;
+  border-bottom: ${foreground || $design.accentForeground} solid 2px;
```
Comment (ZHollingshead) [inline src/tabs/ux-tab-theme.css]: The indicator should be `$design.accent` if we are going to base this on the accent color. `$design.accentForeground` is not always `#fff`. In addition, I would also like to see `foreground` changed to a new property specific to the indicator that can be set in the .ts theme.

> Indicator
>
> - Height: 2dp
> - Color: #fff or accent color

[Material Guidelines - Tabs](https://material.io/guidelines/components/tabs.html#tabs-specs)

File: src/tabs/ux-tab-theme.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
+styles.tab {
+  display: inline-block;
+  text-transform: uppercase;
+  cursor: pointer;
+  opacity: 0.7;
+  padding: 14px 12px 20px 12px;
+  text-align: center;
+  color: ${foreground || $design.accentForeground};
+  background-color: ${background || $design.accent};
```
Comment (ZHollingshead) [inline src/tabs/ux-tab-theme.css]: I would not base the background on the accent.

Try leaving this as either transparent or set it to the primary color.

**Comment by manuel-guilbault** [comment]
@ZHollingshead so with icons, it would give something like this:

```html
<ux-tabs>
  <ux-tab><ux-icon icon="face"></ux-icon> Account</ux-tab>
  <ux-tab><ux-icon icon="favorite"></ux-icon>Favorites</ux-tab>
</ux-tabs>
```

Is this what you have in mind?

**Comment by ZHollingshead** [comment]
I was thinking just for text content. I like how the icon is rendered it feels clean. This is more as an optional way to add text or content to a tab. So one could either do `<ux-tab>Text</ux-tab>` or `<ux-tab text="Text"></ux-tab>`. I hope that makes sense.

This should be achievable by adding a slot. If the below snippet does not work then just move the binding outside of the slot, I cannot remember off the top of my head if you can use a binding as default slot content.

In src/tabs/ux-tab.html
``` html
<span if.bind="text" class="text"><slot>${text}</slot></span>
```

So I am thinking good usage could be:
``` html
<ux-tabs>
  <ux-tab>Account</ux-tab>
  <ux-tab>Favorites</ux-tab>
</ux-tabs>

<ux-tabs>
  <ux-tab icon="face"> Account</ux-tab>
  <ux-tab icon="favorite">Favorites</ux-tab>
</ux-tabs>

<ux-tabs>
  <ux-tab icon="face" text="Account"></ux-tab>
  <ux-tab icon="favorite" text="Favorites"></ux-tab>
</ux-tabs>
```

**Comment by manuel-guilbault** [comment]
Ok makes total sense. I'll update my pull request in a couple of minutes.

On another topic, I would like to animate the indicator when the selected tab changes, but I'll need some help with that, as I don't know much about low-level animations (I always used libraries like jQuery or Velocity). Any idea on how this could be implemented? Any resources which could point me in the right direction?

**Comment by ZHollingshead** [comment]
For animation take a look at these. They are not bad for getting started with animation. One thing to keep in mind is always using `transform` based animations, as those take advantage of hardware acceleration.

https://css-tricks.com/almanac/properties/a/animation/
https://css-tricks.com/almanac/properties/t/transform/

**Comment by manuel-guilbault** [comment]
I just pushed some updates. Let me know if there's anything I forgot or misunderstood.

Thanks for the links. I'll give it a look and let you know how it goes :)

**Comment by manuel-guilbault** [comment]
@ZHollingshead I just pushed a first attempt at animating the indicator. It works well, except for a weird glitch: when the showcase app loads, there's a small duration (a fraction of a second) during which the whole app seems to be rendered bigger, then every element on the page shrinks in size. Because of this, the indicator is initially rendered too big. I don't understand what is the cause of this. I'll keep digging on my end, but if you have any idea of why this happens, let me know ;)

**Pull request merged**